### ISSUE TITLE
Support creation and sharing of image annotations

### ISSUE THREAD

**Issue by kevinyamauchi** [napari/napari-core#20]
I am not sure if this is a core feature or plugin, but it would be really nice if there was a way to annotate features in images (e.g., "check out this interesting blob on this cell!") and share them with collaborators. I think this is useful both for sharing observations and recording them for later analysis. I suppose an annotation mechanism could also be used to label data for training machine learning algorithms.

Perhaps the annotations can be stored in something like [GeoJSON](http://geojson.org/) in the metadata and then they can be browsed, selected for display, and overlaid in the GUI.?

If people think this would be useful, I am happy to do some prototyping because I have been wanting to create something like this at the Biohub anyway.

**Comment by kevinyamauchi** [comment]
[lableme](https://github.com/wkentaro/labelme) looks like it has a lot of the functionality and it uses Qt for the GUI.

**Comment by royerloic** [comment]
I guess this would use the ‘object layer’ concept where non-image layers 
consisting of points, lines, circle, ellipses, rectangles, etc…
can be overlaid over an image…
Conceptually this is very straightforward I think…

Collaborative editing is a bit more challenging if one want to do this.
But then the czbiohub imaging database simply needs to support storing
such layers…

**Comment by kevinyamauchi** [comment]
I agree that the main requirement would be support for an "object layer', which I think will be useful for many things in addition to these annotation. 

For now, I think just saving and recalling the annotations is more than sufficient. Of course, truly collaborative editing would be an awesome feature in the future!
